SimCalorimetry/EcalSimAlgos/interface/EcalTDigitizer_Ph2.h

```diff
@@ -0,0 +1,61 @@
+#ifndef SimCalorimetry_EcalSimAlgos_EcalTDigitizer_Ph2_h
+#define SimCalorimetry_EcalSimAlgos_EcalTDigitizer_Ph2_h
+
+/** Turns hits into digis.  Assumes that 
+    there's an ElectroncsSim class with the
+    interface analogToDigital(CLHEP::HepRandomEngine*, const CaloSamples &, Digi &);
+*/
+
+#include "SimDataFormats/CrossingFrame/interface/MixCollection.h"
+#include "SimDataFormats/CaloHit/interface/PCaloHit.h"
+#include "DataFormats/DetId/interface/DetId.h"
+#include "CalibFormats/CaloObjects/interface/CaloTSamplesBase.h"
+#include "SimCalorimetry/EcalSimAlgos/interface/EcalBaseSignalGenerator.h"
+
+class EcalHitResponse;
+class EcalBaseSignalGenerator;
+
+namespace CLHEP {
+  class HepRandomEngine;
+}
+
+template <class Traits>
+class EcalTDigitizer_Ph2 {
```

Comment:
the entire point of making this a class template is to merge it with EcalTDigitizer and reduce unnecessary duplication